### PR TITLE
Adding a test plan for QUARKUS-713 Tech preview support for the OpenShift Client.

### PR DESCRIPTION
**Pull request by mjurc**

### CODE REVIEW

File: QUARKUS-713.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# QUARKUS-713 Tech preview support for the OpenShift Client
+
+This document describes a high-level testing overview of [QUARKUS-713 Tech preview support for the OpenShift Client](https://issues.redhat.com/browse/QUARKUS-713).
```
Comment (kshpak) [inline QUARKUS-713.md:3]:
```suggestion
JIRA link: https://issues.redhat.com/browse/QUARKUS-713
```
To follow the common structure in all test plans

File: QUARKUS-713.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+The goal of QUARKUS-713 is to provide the users with tech preview support of utility that will expose an OpenShift
+client. This utility will make it possible for the users to develop their kubectl plugins or OpenShift operators.
```
Comment (kshpak) [inline QUARKUS-713.md:6]:
```suggestion
The goal of this test plan is to provide users with the tech preview support of the utility that exposes an OpenShift
client. This utility will make it possible for users to develop their own **kubectl** plugins or OpenShift **operators**.
```

File: QUARKUS-713.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Scope of the testing
+### Smoke testing
+[`OpenShiftClientTest`](https://github.com/quarkusio/quarkus/blob/master/integration-tests/openshift-client/src/test/java/io/quarkus/it/openshift/client/OpenShiftClientTest.java) 
+was implemented as part of feature development. This unit test verifies the basic functional assertions in a mock 
+environment.
+
+This test is executed upstream as a part of Quarkus test suite.
```
Comment (kshpak) [inline QUARKUS-713.md:14]:
```suggestion
[`OpenShiftClientTest`](https://github.com/quarkusio/quarkus/blob/master/integration-tests/openshift-client/src/test/java/io/quarkus/it/openshift/client/OpenShiftClientTest.java) 
was implemented as part of the feature development. This unit test verifies the basic functional assertions in a mock environment.

This test is executed upstream as a part of the Quarkus test suite.
```

File: QUARKUS-713.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+### Component testing
+The feature itself consists of exposing fabric8 OpenShift client to the Quarkus application conveniently.
+
+fabric8 itself has an [upstream CI](https://github.com/fabric8io/kubernetes-client/actions) where the functionality of 
+all the client libraries is verified.
```
Comment (kshpak) [inline QUARKUS-713.md:20]:
```suggestion
The feature consists of exposing the fabric8 OpenShift client to the Quarkus application.
**fabric8** has an [upstream CI](https://github.com/fabric8io/kubernetes-client/actions), where the functionality of 
all the client libraries is verified.
```

File: QUARKUS-713.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+### Platform testing
+The tests will be executed on the following platforms:
+
+* OCP 3.11
+* Latest LTS release of OpenShift 4
+* Latest rolling release of OpenShift 4
+
+## Getting familiar with the feature
+Getting introduced to OpenShift operator development and upstream documentation of the Quarkus feature was used as a
+starting point to develop the initial ideas and implementation of simple operators.
```
Comment (kshpak) [inline QUARKUS-713.md:38]:
```suggestion
Upstream documentation of the Quarkus feature and an introduction to OpenShift operator development was used as a
starting point to develop the initial ideas and implementation of simple operators.
```

File: QUARKUS-713.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+fabric8 itself has an [upstream CI](https://github.com/fabric8io/kubernetes-client/actions) where the functionality of 
+all the client libraries is verified.
+
+### Integration testing
+An integration test verifying correctness of the feature on OpenShift clusters will be implemented in 
+[quarkus-qe/quarkus-openshift-test-suite](https://github.com/quarkus-qe/quarkus-openshift-test-suite).
+
+The test will consist of a simple operator implemented with the new feature. The operator will watch `configMap` 
+resources in the test namespace and generate a deployment.
```
Comment (kshpak) [inline QUARKUS-713.md:27]:
```suggestion
### Impact on test suites and testing automation:
 - The new integration test will be implemented in 
[quarkus-qe/quarkus-openshift-test-suite](https://github.com/quarkus-qe/quarkus-openshift-test-suite) to verify the correctness of the feature on OpenShift clusters.

The test will consist of a simple operator implemented with the new feature. The operator will watch `configMap` 
resources in the test namespace and generate a deployment.
```

File: QUARKUS-713.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+Getting introduced to OpenShift operator development and upstream documentation of the Quarkus feature was used as a
+starting point to develop the initial ideas and implementation of simple operators.
+
+## Contacts
+* Tester: Michal Jurč <mjurc@redhat.com>
+
+## References
+* Feature issue: [QUARKUS-713 Tech preview support for the OpenShift Client](https://issues.redhat.com/browse/QUARKUS-713).
+* Upstream feature issue: [Implement OpenshiftClient on Kubernetes Client Extension #3200](https://github.com/quarkusio/quarkus/issues/3200)
+* Upstream documentation: [Quarkus - Kubernetes Client # OpenShift client](https://quarkus.io/guides/kubernetes-client#openshift-client)
```
Comment (kshpak) [inline QUARKUS-713.md:46]: Get rid of this chapter.
Place JIRA link and GH link on the top. Upstream doc can be added as a hyperlink inside "Getting familiar with the feature"

File: QUARKUS-713.md (hunk 1 above)
Comment (pjgg) [reply QUARKUS-713.md:3]: Well, I like the @mjurc brief description.

File: QUARKUS-713.md (hunk 2 above)
Comment (pjgg) [reply QUARKUS-713.md:6]: Maybe we could remove "the users".
```suggestion
The goal of QUARKUS-713 is to provide tech preview support for the OpenShift
client library. This utility will make it possible to develop their kubectl plugins or OpenShift operators.
```

File: QUARKUS-713.md (hunk 6 above)
Comment (pjgg) [reply QUARKUS-713.md:27]:
```suggestion
### Impact on test suites and testing automation:

integration test will be implemented in ...
```

**Review by pjgg — CHANGES_REQUESTED** [review]

File: QUARKUS-713.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+### Component testing
+The feature itself consists of exposing fabric8 OpenShift client to the Quarkus application conveniently.
+
+fabric8 itself has an [upstream CI](https://github.com/fabric8io/kubernetes-client/actions) where the functionality of 
+all the client libraries is verified.
+
+### Integration testing
+An integration test verifying correctness of the feature on OpenShift clusters will be implemented in 
+[quarkus-qe/quarkus-openshift-test-suite](https://github.com/quarkus-qe/quarkus-openshift-test-suite).
+
```
Comment (rsvoboda) [inline QUARKUS-713.md:25]: Please add some details about current usage of `OpenShiftClient` in OpenShiftTS (outside the OpenShift cluster scenario).

Add explicit note that the new test coverage is about functionality of `OpenShiftClient` inside the OpenShift cluster.

File: QUARKUS-713.md (hunk 7 above)
Comment (mjurc) [reply QUARKUS-713.md:46]: I don't particularly like this suggestion, I mean, how are references hurting anything? This is convenient - all the existing stuff is in a single, stylistically expected place. I don't think a document should start with references, either.

File: QUARKUS-713.md (hunk 1 above)
Comment (mjurc) [reply QUARKUS-713.md:3]: I'm gonna keep this one, a brief description never hurt anything :)

**Comment by mjurc** [comment]
@rsvoboda @pjgg @kshpak I've rewrote this in the briefer format the TPs in this repo use. I'd kindly ask for a review, thanks

File: QUARKUS-713.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+- Define the scope of the testing
+- Analyze the possible impact on Quarkus itself - e.g performance, compatibility
+- Determine impact on test suites - e.g. new test suite, increased complexity of existing test suite, maintenance burden
+- Determine impact on QE infrastructure - e.g. increased complexity of existing automation, maintenance burden, need for new external resources
+
+### Impact on testsuites and testing automation:
+- Upstream smoke test [`OpenShiftClientTest`](https://github.com/quarkusio/quarkus/blob/master/integration-tests/openshift-client/src/test/java/io/quarkus/it/openshift/client/OpenShiftClientTest.java)
+  was implemented as part of feature development
+- fabric8 client libraries is verified in the [upstream CI](https://github.com/fabric8io/kubernetes-client/actions)
```
Comment (kshpak) [inline QUARKUS-713.md:19]:
```suggestion
- fabric8 client libraries are verified in the [upstream CI](https://github.com/fabric8io/kubernetes-client/actions)
```

File: QUARKUS-713.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+- [quarkus-qe/quarkus-openshift-test-suite](https://github.com/quarkus-qe/quarkus-openshift-test-suite) uses the fabric8 OpenShift client to manage kube objects from test code
+- New integration test will be implemented [quarkus-qe/quarkus-openshift-test-suite](https://github.com/quarkus-qe/quarkus-openshift-test-suite)
```
Comment (kshpak) [inline QUARKUS-713.md:21]:
```suggestion
- New integration test will be implemented in [quarkus-qe/quarkus-openshift-test-suite](https://github.com/quarkus-qe/quarkus-openshift-test-suite)
```

**Review by kshpak — CHANGES_REQUESTED** [review]
Besides mentioned, LGTM

File: QUARKUS-713.md (hunk 7 above)
Comment (rsvoboda) [reply QUARKUS-713.md:39]: I have no strong feeling about having summary of the links in "References"

JIRA and GH link are (now?) at the top too, so I'm fine with current format

**Comment by rsvoboda** [comment]
Nothing new (comparing to Kyrylo's response 3 hours ago) from my side.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**